### PR TITLE
Add documentation for celltype annotation

### LINKED ISSUE #130: Document cell type annotations
We will need to update both the processing information doc and the SCE contents documents to include information about the available cell type annotations.

### PR DESCRIPTION
**Pull request by sjspielman**
Closes #130 

This PR adds docs for cell typing as described in #130, largely based on the comments I wrote after the issue was opened. 
Let me know where I can tweak any phrasing, what can use more or less detail, etc. Also, anything I am entirely missing?

### CODE REVIEW

File: docs/processing_information.md [hunk 1]
```diff
@@ -71,6 +71,19 @@ Finally, these principal components are used to calculate the [UMAP (Uniform Man
 
 [Graph-based clustering](http://bioconductor.org/books/3.16/OSCA.basic/clustering.html#clustering-graph) is also performed, using the Louvain algorithm with 20 nearest neighbors and Jaccard weighting.
 
+#### Cell type annotation
+
+We additional perform cell type annotation with two complementary methods, where possible:
```
Comment (allyhawkins) [inline docs/processing_information.md:76]:
```suggestion
We perform cell type annotation with two complementary methods, where possible:
```

File: docs/processing_information.md [hunk 2]
```diff
@@ -71,6 +71,19 @@ Finally, these principal components are used to calculate the [UMAP (Uniform Man
...
+
+- [`SingleR`](https://bioconductor.org/packages/release/bioc/html/SingleR.html), a reference-based cell type annotation method
+- [`CellAssign`](https://github.com/Irrationone/cellassign), a marker-gene--based cell type annotation method
+
+For `SingleR` annotation, we identify an appropriate reference dataset from the [`celldex` package](http://bioconductor.org/packages/release/data/experiment/html/celldex.html) and train the classification model to use ontology IDs for annotation.
+Cells which `SingleR` cannot be confidently are given `NA` annotations.
```
Comment (allyhawkins) [inline docs/processing_information.md:82]:
```suggestion
Cells which `SingleR` cannot confidently assign are given `NA` annotations.
```

File: docs/processing_information.md [hunk 3]
```diff
@@ -71,6 +71,19 @@ Finally, these principal components are used to calculate the [UMAP (Uniform Man
...
+
+For `CellAssign` annotation, we identify an appropriate set of marker genes for the given tissue type from the [`PanglaoDB`](https://panglaodb.se/) database.
+During annotation, we additionally include an `"other"` cell type does not express any of these marker genes.
```
Comment (allyhawkins) [inline docs/processing_information.md:85]:
```suggestion
During annotation, we additionally include an `"other"` cell type that does not express any of these marker genes.
```

File: docs/processing_information.md [hunk 4]
```diff
@@ -71,6 +71,19 @@ Finally, these principal components are used to calculate the [UMAP (Uniform Man
...
+
+We additional perform cell type annotation with two complementary methods, where possible:
+
+- [`SingleR`](https://bioconductor.org/packages/release/bioc/html/SingleR.html), a reference-based cell type annotation method
+- [`CellAssign`](https://github.com/Irrationone/cellassign), a marker-gene--based cell type annotation method
+
+For `SingleR` annotation, we identify an appropriate reference dataset from the [`celldex` package](http://bioconductor.org/packages/release/data/experiment/html/celldex.html) and train the classification model to use ontology IDs for annotation.
+Cells which `SingleR` cannot be confidently are given `NA` annotations.
+
+For `CellAssign` annotation, we identify an appropriate set of marker genes for the given tissue type from the [`PanglaoDB`](https://panglaodb.se/) database.
```
Comment (allyhawkins) [inline docs/processing_information.md:84]: We are also combining these with an immune organ, so I think we want to point that out too.

File: docs/sce_file_contents.md [hunk 5]
```diff
@@ -57,12 +57,16 @@ See the description of the {ref}`processed gene expression data <processing_info
...
 
-The `processed` object has one additional `colData` column reflecting cluster assignments:
+The `processed` object has one additional `colData` column reflecting cluster assignments.
+Further, if cell type annotation was performed, there will be additional columns representing annotation results in the `processed` object's `colData`.
 
 | Column name             | Contents                                              |
 | ----------------------- | ----------------------------------------------------- |
 | `clusters`  | Cell cluster identity identified by graph-based clustering |
-
+| `singler_celltype_annotation`  | If cell typing with `SingleR` was performed, the annotated cell type |
```
Comment (allyhawkins) [inline docs/sce_file_contents.md:66]: I think here and in the `cellassign_celltype_annotation`, I would note what the other/NA labels are again.

File: docs/sce_file_contents.md [hunk 6]
```diff
@@ -57,12 +57,16 @@ See the description of the {ref}`processed gene expression data <processing_info
 | `scpca_filter` | Labels cells as either `Keep` or `Remove` based on filtering criteria (`prob_compromised` < 0.75 and number of unique genes detected > 200) |
 | `adt_scpca_filter` | If CITE-seq was performed, labels cells as either `Keep` or `Remove` based on ADT filtering criteria (`discard = TRUE` as determined by [`DropletUtils::CleanTagCounts()`](https://rdrr.io/github/MarioniLab/DropletUtils/man/cleanTagCounts.html)) |
 
-The `processed` object has one additional `colData` column reflecting cluster assignments:
+The `processed` object has one additional `colData` column reflecting cluster assignments.
+Further, if cell type annotation was performed, there will be additional columns representing annotation results in the `processed` object's `colData`.
```
Comment (allyhawkins) [inline docs/sce_file_contents.md:61]: I would add a link to the processing information to get more information about the cell typing.

File: docs/sce_file_contents.md [hunk 7]
```diff
@@ -116,6 +120,12 @@ expt_metadata <- metadata(sce)
 | `cluster_algorithm` | The algorithm used to perform graph-based clustering of cells. Only present for `processed` objects |
 | `cluster_weighting` | The weighting approach used during graph-based clustering. Only present for `processed` objects |
 | `cluster_nn`        | The nearest neighbor parameter value used for the graph-based clustering. Only present for `processed` objects |
+| `celltype_methods` | If cell type annotation was performed, a vector of the methods used for annotation. May include `"singler"` and/or `"cellassign"`. Only present for `processed` objects |
+| `singler_results` | If cell typing with `SingleR` was performed, the full result object returned by `SingleR` annotation. Only present for `processed` objects |
+| `singler_reference_name` | If cell typing with `SingleR` was performed, the name of the reference dataset used for annotation. Only present for `processed` objects |
```
Comment (allyhawkins) [inline docs/sce_file_contents.md:125]: I think I would link to celldex here because we are only planning on using celldex references.

**Review by allyhawkins — COMMENTED** [review]
This generally looks good, just a few small comments and additions.

**Comment by sjspielman** [comment]
Hm, what's the deal with this `readthedocs` failure? I definitely did not delete our config file..?

**Comment by sjspielman** [comment]
Github is a mess right now, so who knows if this comment goes through...but I'm actually wondering if this build fail is not our fault and somehow related to github actions being a mess. Indeed, our config is already at v2 syntax..? Let's wait and see here!

**Comment by sjspielman** [comment]
Alright, we're all set with a fixed config situation in #141, back on track!

**Review by allyhawkins — APPROVED** [review]
Looks good! Once this passes checks this is good to go.

**Comment by sjspielman** [comment]
Noting that https://github.com/AlexsLemonade/scpca-docs/pull/134/commits/66d2a7f9873a9921912a202dcd8cbfe1836748aa fixes a silly oversight in #141 where the config file was copied instead of moved..! Hopefully this all now builds!

**Pull request merged**